kernel/src/kernel.rs

```diff
@@ -87,6 +87,34 @@ pub enum StoppedExecutingReason {
     KernelPreemption,
 }
 
+/// Represents the different outcomes when trying to allocation a grant region
+enum AllocResult {
+    NoAlloc,
+    NewAlloc,
+    SameAlloc,
+}
+
+/// Tries to allocate the grant region for specified driver and process.
+/// Returns if a new grant was allocated or not
+fn try_allocate_grant<KR: KernelResources<C>, C: Chip>(
+    resources: &KR,
+    driver_number: usize,
+    process: &dyn process::Process,
+) -> AllocResult {
+    use AllocResult::*;
```

Comment:
Tock style generally forbids wildcard imports, as they can pollute namespaces. For whatever reason clippy does not seem to be detecting this one, but I think we would still prefer to avoid its use here

Reply:
Fair enough, done. I think this is a really good use of the wildcard since the wildcard use is scoped to this small function. I am happy to follow the existing style though. Done.